packages/curve-ui-kit/src/features/connect-wallet/lib/ConnectionContext.tsx

```diff
@@ -0,0 +1,178 @@
+import { ethers } from 'ethers'
+import { createContext, type ReactNode, useContext, useEffect, useRef, useState } from 'react'
+import { getWalletChainId, getWalletSignerAddress, useSetChain, useWallet } from '@ui-kit/features/connect-wallet'
+import { WalletNameStorageKey } from '@ui-kit/features/connect-wallet/lib/hooks'
+import { withTimeout } from '@ui-kit/features/connect-wallet/lib/utils/wallet-helpers'
+import { getFromLocalStorage, setLocalStorage } from '@ui-kit/hooks/useLocalStorage'
+import type { WalletState as Wallet } from '@web3-onboard/core'
+
+export const CONNECT_STATUS = {
+  LOADING: 'loading',
+  SUCCESS: 'success',
+  FAILURE: 'failure',
+}
+
+export const CONNECT_STAGE = {
+  CONNECT_API: 'api',
+  HYDRATE: 'hydrate',
+  CONNECT_WALLET: 'connect-wallet',
+  DISCONNECT_WALLET: 'disconnect-wallet',
+  SWITCH_NETWORK: 'switch-network',
+} as const
```

Comment:
I'm seeing double here with `'stage: 'api' | 'connect-wallet' | 'switch-network' | 'disconnect-wallet' | 'hydrate' | ''` in `utilsConnectstate.tsx`

EDIT: Ah I see, this connection provider isn't used everywhere yet so it's a temporary thing.